src/odemis/driver/hamamatsurx.py

```diff
@@ -0,0 +1,2043 @@
+                    except (TypeError, ValueError, IOError):
+                        logging.warning("Received response, which is not according to the known protocol.")
+                        continue  # return to try-statement and start receiving again
+
+                    if self._getReadoutCamInfo:
+                        # command parent.CamParamGet("Setup", "CameraInfo") behaves differently then all other commands
+                        # nasty trick to work around for this command
+                        # This command is nasty as it first receives the EC and then additional information
+                        # TODO are there more cases like that??
+
+                        # info_size = 2 * 1000  # TODO how many bytes??
+                        # additional_info = []
+                        # time.sleep(1)
+                        # self.timeout_commandport = 10
+                        # self._commandport.settimeout(5.0)
+
+                        # TODO would like to use that, but socket.timeout not handled correctly
+                        # while len(additional_info) < info_size:
+                        #     try:
+                        #         additional_info.append(self._commandport.recv(4096))  # receive more data
+                        #     except socket.timeout:
+                        #         break
+
+                        additional_info = ""
+                        timeout = 1
+                        start = time.time()
+                        time_now = time.time()
+                        while time_now < start + timeout:
+                            try:
+                                # continue listening as there is additional info in coming
+                                additional_info += self._commandport.recv(4096)   # receive more data
+                                time_now = time.time()
+                            except Exception:
+                                break
+                        try:
+                            additional_info = additional_info.split("\r")[:-1]
+                            for i, item in enumerate(additional_info):
+                                rargs.append(item.replace("\n", ""))
+                            msg_splitted[2] = rargs
+                        except Exception:
+                            logging.error("Could not retrieve readout camera information.")
+                        self._getReadoutCamInfo = False
+
+                    if EC in (4, 5):
+                        logging.debug("Received message %s from RemoteEx software." % rargs)
+                        if EC == 4 and rfunc == "Livemonitor":
+                            # TODO maybe check if rargs is empty, should not for type 4 and 5 i think
+                            # if len(rargs) > 0:
+                            self.queue_img.put(rargs)  # only put msg in queue when it notifies about an image
+
```

Comment:
Add a comment here that every other message 4 or 5 is just discarded because currently we are not interested.